src/state/network/updater.tsx

```diff
@@ -33,6 +35,7 @@ export const NetworkUpdater: React.FC = () => {
     // Update the network if it's different
     if (currentNetworkId !== networkId) {
       dispatch(setNetwork(networkId))
+      updateWeb3Provider(web3, networkId)
```

Comment:
mmm, it works, it's even kind of nice, but I think it has a problem.

This state `network` is super specific, probably I should have called that `network-from-path`
It's an opt-in functionality apps can just add in their combine reducer (+add the updater) and benefit from it. 

You are assuming here all users of this functionality will use a web3 wallet.
- What if don't have wallet, and I don't want one
- What if we want to use EtherJS :) 
- What our API of web3 gets deprecated in favour of another one better, but we don't want to adapt all the apps.

Another approach is to add an updater for the API, this updater is simple. It watches the global state for change.
Then, if u need just the `network-from-path` you just use that, if you need the wallet to, you add also the updater for that.

Reply:
> You are assuming here all users of this functionality will use a web3 wallet.

Well, not really. This change is precisely for when the users don't have a connected wallet, and the app needs to change the network based on the currently selected network.

I expect that when users connect a wallet, it'll provide us with the appropriated provider. 
Although, we are not there yet, so not really thinking about this =P

> Another approach is to add an updater for the API, this updater is simple. It watches the global state for change.
Then, if u need just the network-from-path you just use that, if you need the wallet to, you add also the updater for that.

So you suggest a global state for `web3`, that depends on the `network` state?

---

I'm not gonna defend this as the perfect solution, as I'm not too happy with this either.
Happy to do whatever makes more sense.

Reply:
My point is that here you assume there's a Web3Provider provider.

My network state was isolated, not related to web3provider in any way. So now you assume all the apps need to call `updateWeb3Provider` when the `networkId` changes

For me, this is a side effect on the network changing, I'll probably make a more specific suggestion. So forget for now